src/common/hooks/useAllowedJobProfiles.js

```diff
@@ -0,0 +1,35 @@
+import { useQuery } from 'react-query';
+
+import {
+  useOkapiKy,
+  useNamespace,
+} from '@folio/stripes/core';
+
+import {
+  DATA_IMPORT_JOB_PROFILES_ROUTE,
+  LIMIT_MAX,
+} from '../../constants';
+
+const useAllowedJobProfiles = (allowedJobProfileIds) => {
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace({ key: 'allowedJobProfiles' });
+
+  const DATA_TYPE_MARC_QUERY = 'dataType==("MARC")';
+
+  const ids = allowedJobProfileIds.join(' or ');
+  const allowedIdsQuery = `id==(${ids})`;
+  const path = `${DATA_IMPORT_JOB_PROFILES_ROUTE}?limit=${LIMIT_MAX}&query=${DATA_TYPE_MARC_QUERY} and ${allowedIdsQuery} sortBy name`;
+
+  const { isLoading, data: allowedJobProfiles = {} } = useQuery({
+    queryKey: [namespace, allowedJobProfileIds],
+    queryFn: () => ky.get(path).json(),
+    enabled: !!allowedJobProfileIds,
```

Comment:
This array can be empty but you still will get 'true' so you need to check that it's not empty or just check your 'ids' string